### ISSUE TITLE
Exception verification in junit tests

### ISSUE THREAD

**Issue by yrmk** [powermock/powermock#806]
**What steps will reproduce the problem?**
```
@Test(expected = NotFoundException.class)
public void a() {
    doThrow(NotFoundException.class).when(b).doSomething();
    b.doSomething();
}
```
**What is the expected output?**
Test to pass
**What do you see instead?**
Test failure, throws RuntimeExceptionProxy instead of NotFoundException
**What version of the product are you using?**
1.7.0RC4
**On what operating system?**
macOS 10.12.5
**Please provide any additional information below.** 
I believe the issue is with this change:
https://github.com/powermock/powermock/commit/a89e8e5760f57c6d1b92b9049a44d43da92150da#diff-d9ef7e539426f80c697a03a7f7e2c8e9R25

**Comment by thekingnothing** [comment]
Hi @yrmk 

Do you `Mockito` or `PowerMockito` for `doThrow` ?

**Comment by yrmk** [comment]
Hi @thekingnothing
 
Mockito.

However, the problem seems to be whenever any exception is thrown, even if not mocked. It is a problem with all exceptions. The exception always gets wrapped, and so what's thrown isn't the exception itself. Consider this class for instance:

```
public class Foo {
    public void throwExceptionForInput(String foo) throws BarException {
        if ("123".equals(foo)) {
            throw new BarException();
        }    
    }
}
```
And then consider a spied instance of this class named `foo` being tested:
```
@Test(expected=BarException.class)
public void testSpiedFoo() {
    //foo is a spied instance of type Foo
    foo.throwExceptionForInput("123"); //should expect to throw BarException here, but throws RuntimeExceptionProxy instead
}
```

The problem can get nested too, for instance:
 - a certain class has Methods A and B (B throws a certain exception)
 - Method A calls Method B
 - Method A has logic that depends on catching that exception from Method B
 - We test Method A 
 - Because the exception from Method B will always get wrapped as a different exception, A never catches that exception, that logic will never execute, test will fail

**Comment by thekingnothing** [comment]
I've added a test cases for your case and all of them are passed on the current code base. So I guess your issue was already fixed in PowerMock 1.7.0. I'm going to release it soon.
